Tests/SonarQube.MSBuild.Tasks.UnitTests/WriteProjectInfoFileTests.cs

```diff
@@ -220,20 +222,415 @@ public void WriteProjectInfoFile_UnrecognisedLanguages()
+                        return expectedEncoding;
+                    }
+                    else
+                    {
+                        Assert.Fail("should not have been called");
+                        throw new InvalidOperationException();
+                    }
+                },
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                });
+
+            // Act
+            var actual = WriteProjectInfoFile_ExecuteAndReturn(encodingProvider, expectedEncoding.CodePage.ToString(), "whatever", "foo");
+
+            // Assert
+            Assert.AreEqual(actual.Encoding, expectedEncoding.WebName, "unexpected encoding");
+        }
+
+        [TestMethod]
+        public void WriteProjectInfoFile_Execute_WhenCSharpWithNullCodePageAndDefaultSystemEncodingExists_ExpectsDefaultSystemEncoding()
+        {
+            // Arrange
+            var expectedEncoding = Encoding.GetEncoding(1141);
+            var encodingProvider = new TestEncodingProvider(
+                x =>
+                {
+                    if (x == 0)
+                    {
+                        return expectedEncoding;
+                    }
+                    else
+                    {
+                        Assert.Fail("should not have been called");
+                        throw new InvalidOperationException();
+                    }
+                },
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                });
+
+            // Act
+            var actual = WriteProjectInfoFile_ExecuteAndReturn(encodingProvider, null, "C#", "foo2");
+
+            // Assert
+            Assert.AreEqual(actual.Encoding, expectedEncoding.WebName, "unexpected encoding");
+        }
+
+        [TestMethod]
+        public void WriteProjectInfoFile_Execute_WhenCSharpVBNetWithInferiorToZeroCodePageAndDefaultSystemEncodingExists_ExpectsDefaultSystemEncoding()
+        {
+            // Arrange
+            var expectedEncoding = Encoding.GetEncoding(1141);
+            var encodingProvider = new TestEncodingProvider(
+                x =>
+                {
+                    if (x == 0)
+                    {
+                        return expectedEncoding;
+                    }
+                    else
+                    {
+                        Assert.Fail("should not have been called");
+                        throw new InvalidOperationException();
+                    }
+                },
+                x =>
+                {
+                    Assert.Fail("should not have been called");
+                    throw new InvalidOperationException();
+                });
+
+            // Act
+            var actualCSharp = WriteProjectInfoFile_ExecuteAndReturn(encodingProvider, "-1", "C#", "foo31");
```

Comment:
The second argument of this method seems magical: `"-1"`, `null`, and `"42"` are all passed. Let's extract these into constants with proper naming. Also, if it's always an `int?` then use that instead of `string`.